warehouse/query-core/src/main/java/datawave/query/attributes/Numeric.java

```diff
@@ -69,10 +69,23 @@ public Object getData() {
     }
 
     private Number parseToNumber(String value) {
-        Number number = null;
+        // some documents will contain numeric data that is already encoded
+        // in those cases it is faster to check for encoding up front instead
+        // of finding out the hard way via exception.
+        try {
+            if (NumericalEncoder.isPossiblyEncoded(value)) {
+                BigDecimal bigDecimal = NumericalEncoder.decode(value);
+                return bigDecimal.doubleValue();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Should we add a message to this RE, or is the thrown exception specific enough?

If its a specific exception, should we only catch the specific exception rather than Exception?

Reply:
The thrown exception should always be an IllegalArgumentException which is specific enough for me. The NumericalEncoder.decode method catches Exception to throw the RuntimeException so I think replicating that pattern here is also okay.